sanic/mixins/routes.py

```diff
@@ -806,23 +805,23 @@ async def _static_request_handler(
         content_type=None,
         __file_uri__=None,
     ):
-        # Using this to determine if the URL is trying to break out of the path
-        # served.  os.path.realpath seems to be very slow
-        if __file_uri__ and "../" in __file_uri__:
-            raise BadRequest("Invalid URL")
         # Merge served directory and requested file if provided
-        # Strip all / that in the beginning of the URL to help prevent python
-        # from herping a derp and treating the uri as an absolute path
-        root_path = file_path = file_or_directory
+        root_path = file_path = path.abspath(unquote(file_or_directory))
+
         if __file_uri__:
-            file_path = path.join(
-                file_or_directory, sub("^[/]*", "", __file_uri__)
-            )
+            # Strip all / that in the beginning of the URL to help prevent
+            # python from herping a derp and treating the uri as an
+            # absolute path
+            unquoted_file_uri = unquote(__file_uri__).lstrip("/")
+
+            segments = unquoted_file_uri.split("/")
+            if ".." in segments or any(sep in segment for segment in segments):
```

Comment:
we should benchmark this. Running two loops where only one is needed.

Reply:
How about `if any(sep in segment or segment == ".." for segment in segments):`?

Reply:
That looks good to me, although my personal preference would be to place the `segment == ".."` check first, since it's quicker to check. However this is probably an unnecessary micro optimisation and either way works fine :P

Reply:
@ahopkins Do you mean `sep in segment` was another loop or `".." in segments` was another loop?
@prryplatypus Yeah, that make sense.

Reply:
I never had a chance to do the benchmark before. Is there any guide for it?